app/lib/library/src/library.cpp

```diff
@@ -0,0 +1,47 @@
+#include <library.hpp>
+#include <Arduino.h>
+#include <SD.h>
+
+
+const int SDCARD_MOSI_PIN = 11;
+const int SDCARD_SCK_PIN = 13;
+const int SDCARD_CS_PIN = BUILTIN_SDCARD;
+
+// create the array of filenames
+void Library::array(){
+    File root = SD.open("/");

```

Comment:
Reading from the root directory displays the files that are being used as "file buffers", which shouldn't be accessible. We should probably use a different subdirectory, instead. In the branch `feature/save-loop`, a _loops_ subdirectory is used. In any case, we would just check if the directory exists, and create it if not.